stdlib/public/stubs/CommandLine.cpp

```diff
@@ -242,6 +242,39 @@ char ** _swift_stdlib_getUnsafeArgvArgc(int *outArgLen) {
 
   return argv;
 }
+#elif defined(__OpenBSD__)
+#include <sys/types.h>
+#include <sys/sysctl.h>
+#include <sys/exec.h>
+
+SWIFT_RUNTIME_STDLIB_API
+char ** _swift_stdlib_getUnsafeArgvArgc(int *outArgLen) {
+  if (_swift_stdlib_ProcessOverrideUnsafeArgv) {
+    *outArgLen = _swift_stdlib_ProcessOverrideUnsafeArgc;
+    return _swift_stdlib_ProcessOverrideUnsafeArgv;
+  }
+
+  int mib[2] = {CTL_VM, VM_PSSTRINGS};
+  struct _ps_strings _ps;
+  size_t len = sizeof(_ps);
+  if (sysctl(mib, 2, &_ps, &len, NULL, 0) != 0) {
+    return nullptr;
+  }
+
+  struct ps_strings *ps = (struct ps_strings *)_ps.val;
+  *outArgLen = ps->ps_nargvstr;
+
+  std::vector<char *> argv;
```

Comment:
The only reason the other platforms need this is because they don't have a nice API like `ps_nargvstr` to let them know how big the buffer is going to be. You shouldn't need to do this.

Reply:
I was thinking that we wanted to return a copy of the arguments from _swift_stdlib_getUnsafeArgvArgc, rather than the actual arguments themselves. I was also feeling a little cautious, given that `sys/exec.h` says "The ps program uses it to locate argv and environment strings. Programs that wish ps to display other information may modify it; normally ps_argvstr points to argv[0], and ps_nargvstr is the same as the program's argc."

Thinking about this a little more, there doesn't _seem_ to be a contractual requirement for `CommandLine.unsafeArgv` to get a copy, can you confirm?

Reply:
You will need to `strdup` these strings into a buffer, yes. My comment is about the needless vector allocations.